web/src/components/storage/ProposalSpacePolicyField.jsx

```diff
@@ -428,45 +418,137 @@ const SpacePolicyPicker = ({ currentPolicy, onChange = noop }) => {
  * @param {boolean} [isLoading=false]
  * @param {(settings: ProposalSettings) => void} [onChange]
  */
-export default function ProposalSpacePolicySection({
-  settings,
+const SpacePolicyForm = ({
+  id,
+  currentPolicy,
+  currentActions,
+  devices,
   isLoading = false,
-  onChange = noop
-}) {
-  const changeSpacePolicy = (policy) => {
-    onChange({ spacePolicy: policy });
-  };
+  onSubmit = noop
+}) => {
+  const [policy, setPolicy] = useState(currentPolicy);
+  const [actions, setActions] = useState(currentActions);
+  const [customUsed, setCustomUsed] = useState(false);
 
-  const changeSpaceActions = (spaceAction) => {
-    const spaceActions = settings.spaceActions.filter(a => a.device !== spaceAction.device);
+  // The selectors for the space action have to be initialized always to the same value
+  // (e.g., "keep") when the custom policy is selected for first time. The following two useEffect
+  // ensures that.
+
+  // Stores whether the custom policy has been used.
+  useEffect(() => {
```

Comment:
See https://github.com/openSUSE/agama/pull/1090/files#r1526086667. As said there, I'm not sure this will be the final behavior, but it's not a stopper to merge (is actually quite good).